From c02154eee59e3130a793502f59888ddf94c4cbf7 Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Wed, 16 Feb 2022 00:20:09 +0800
Subject: [PATCH 08/50] Revert "drm/msm/hdmi: Fix missing put_device() call in
 msm_hdmi_get_phy"

This reverts commit 0b7d8db87d0bc7af1c27aa62780c2921d69e834a.
---
 drivers/gpu/drm/msm/hdmi/hdmi.c | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/drivers/gpu/drm/msm/hdmi/hdmi.c b/drivers/gpu/drm/msm/hdmi/hdmi.c
index 94f948ef279d..737453b6e596 100644
--- a/drivers/gpu/drm/msm/hdmi/hdmi.c
+++ b/drivers/gpu/drm/msm/hdmi/hdmi.c
@@ -97,15 +97,10 @@ static int msm_hdmi_get_phy(struct hdmi *hdmi)
 
 	of_node_put(phy_node);
 
-	if (!phy_pdev) {
+	if (!phy_pdev || !hdmi->phy) {
 		DRM_DEV_ERROR(&pdev->dev, "phy driver is not ready\n");
 		return -EPROBE_DEFER;
 	}
-	if (!hdmi->phy) {
-		DRM_DEV_ERROR(&pdev->dev, "phy driver is not ready\n");
-		put_device(&phy_pdev->dev);
-		return -EPROBE_DEFER;
-	}
 
 	hdmi->phy_dev = get_device(&phy_pdev->dev);
 
-- 
2.27.0

